subprojects/GOPATH/src/github.com/microsoft/hivedscheduler/pkg/algorithm/cell.go

```diff
@@ -40,14 +41,34 @@ type Cell interface {
 	SetParent(Cell)
 	GetChildren() CellList
 	SetChildren(CellList)
+	AtOrHigherThanNode() bool
+
+	IncreaseFreeGpuNum(int32)
+	GetFreeGpuNumAtPriority(CellPriority) int32
+	GetUsedGpuNumAtPriority(CellPriority) int32
+	IncreaseUsedGpuNumAtPriority(CellPriority, int32)
+	GetUsedGpuNumSamePriority() int32
+	GetUsedGpuNumOtherPriority() int32
+	UpdateUsedGpuNumForPriority(CellPriority)
+	UpdateUsedGpuNumAllPriority(CellPriority)
 }
 
 type GenericCell struct {
-	chain    CellChain
-	level    CellLevel
-	priority CellPriority // priority of a cell is max of those of its children and itself
-	parent   Cell         // pointer to its parent cell
-	children CellList     // pointer to its children cells
+	chain              CellChain
+	level              CellLevel
+	priority           CellPriority
+	parent             Cell     // pointer to its parent cell
+	children           CellList // pointer to its children cells
+	atOrHigherThanNode bool     // true if the cell is at or higher than node level
+
+	totalGpuNum          int32                  // total GPU number of a cell
+	freeGpuNum           int32                  // free GPU number of a cell
+	freeGpuNumAtPriority int32                  // free GPU number at a certain priority (lower priority considered as free)
```

Comment:
>certain priority [](start = 69, length = 16)

Is it the "priority as that of the pod to be scheduled"? Maybe we should have a better name.

Reply:
yes, it is. I refined the comment.

---
In reply to: [334401987](https://github.com/microsoft/pai/pull/3684#discussion_r334401987) [](ancestors = 334401987)